.gitlab/package_build/deb.yml

```diff
@@ -41,86 +41,131 @@
+      "go_deps",
+      "generate_minimized_btfs_x64",
+    ]
   variables:
     AGENT_MAJOR_VERSION: 6
-    PYTHON_RUNTIMES: '2,3'
+    PYTHON_RUNTIMES: "2,3"
     PACKAGE_ARCH: amd64
-    DESTINATION_DEB: 'datadog-agent_6_amd64.deb'
-    DESTINATION_DBG_DEB: 'datadog-agent-dbg_6_amd64.deb'
+    DESTINATION_DEB: "datadog-agent_6_amd64.deb"
+    DESTINATION_DBG_DEB: "datadog-agent-dbg_6_amd64.deb"
   before_script:
     - export RELEASE_VERSION=$RELEASE_VERSION_6
 
 agent_deb-x64-a7:
   extends: .agent_build_common_deb
   rules:
-    !reference [.on_a7]
+    - !reference [.on_omnibus_change]
+    - !reference [.on_go-version_change]
+    - !reference [.on_kitchen_test_change]
+    - !reference [.on_a7]
   stage: package_build
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
   tags: ["arch:amd64"]
-  needs: ["go_mod_tidy_check", "build_system-probe-x64", "go_deps", "generate_minimized_btfs_x64"]
+  needs:
+    [
+      "go_mod_tidy_check",
+      "build_system-probe-x64",
+      "go_deps",
+      "generate_minimized_btfs_x64",
+    ]
   variables:
     AGENT_MAJOR_VERSION: 7
-    PYTHON_RUNTIMES: '3'
+    PYTHON_RUNTIMES: "3"
     PACKAGE_ARCH: amd64
-    DESTINATION_DEB: 'datadog-agent_7_amd64.deb'
-    DESTINATION_DBG_DEB: 'datadog-agent-dbg_7_amd64.deb'
+    DESTINATION_DEB: "datadog-agent_7_amd64.deb"
+    DESTINATION_DBG_DEB: "datadog-agent-dbg_7_amd64.deb"
   before_script:
     - export RELEASE_VERSION=$RELEASE_VERSION_7
 
 agent_deb-x64-a7-auto:
   extends: .agent_build_common_deb
   rules:
-    !reference [.on_a7]
+    - !reference [.on_omnibus_change]
+    - !reference [.on_go-version_change]
+    - !reference [.on_kitchen_test_change]
+    - !reference [.on_a7]
   stage: package_build
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
   tags: ["arch:amd64"]
-  needs: ["go_mod_tidy_check", "build_system-probe-x64", "go_deps", "generate_minimized_btfs_x64"]
+  needs:
+    [
+      "go_mod_tidy_check",
+      "build_system-probe-x64",
+      "go_deps",
+      "generate_minimized_btfs_x64",
+    ]
   variables:
     AGENT_MAJOR_VERSION: 7
-    PYTHON_RUNTIMES: '3'
+    PYTHON_RUNTIMES: "3"
     PACKAGE_ARCH: amd64
-    DESTINATION_DEB: 'datadog-agent_7_auto_amd64.deb'
-    DESTINATION_DBG_DEB: 'datadog-agent-dbg_7_auto_amd64.deb'
+    DESTINATION_DEB: "datadog-agent_7_auto_amd64.deb"
+    DESTINATION_DBG_DEB: "datadog-agent-dbg_7_auto_amd64.deb"
   before_script:
     - export RELEASE_VERSION=$RELEASE_VERSION_7
     - export INSTALL_DIR=/opt/datadog/agents/$RELEASE_VERSION_7
 
 agent_deb-arm64-a6:
   extends: .agent_build_common_deb
   rules:
-    !reference [.on_all_builds_a6]
+    - !reference [.on_omnibus_change]
+    - !reference [.on_go-version_change]
+    - !reference [.on_kitchen_test_change]
+    - !reference [.on_all_builds_a6]
```

Comment:
Same here I think we should put first `.on_all_build_a6` (same for a7). Because if `RELEASE_VERSION_6` is not defined we must not try to run this job

Reply:
Addressed in 0f4d05da8d